thunder/common.py

```diff
@@ -734,26 +733,10 @@ def _fn(*args, **kwargs) -> tuple[Any, list[TraceCtx]]:
                 tensor_cls = (torch.Tensor, TensorProxy)
                 requires_grad = any(isinstance(arg, tensor_cls) and arg.requires_grad for arg in flat_args)
                 if not cd.disable_torch_autograd_support and requires_grad:
-                    # thunder_backward may recursively call compile and wraps the result in a
-                    # torch.autograd.Function to support embedding of Thunder-compiled
-                    # functions in torch's Autograd
-                    cs.last_trace_host_execution_start = time.time_ns()
-                    c = thunder_backward(compile_data=cd, compile_stats=cs)(processed_function)
-                    result = c(*args, **kwargs)
-                    cs.last_trace_host_execution_stop = time.time_ns()
-                    cs.last_executed = c
-                    if cd.cache_option is CACHE_OPTIONS.CONSTANT_VALUES:
-                        cache_put(
-                            cs.cache,
-                            c,
-                            None,
-                            args[cd.num_constant_args :],
-                            kwargs,
-                            autocast_key=None,
-                            distributed_key=distributed_key,
-                        )
-                    cs.last_trace_host_stop = time.time_ns()
-                    return result
+                    raise NotImplementedError(
+                        "torch.autograd.Function integration is not supported in thunder.compile(). "
+                        "Please use thunder.jit() to compile functions that require torch.autograd.Function."
+                    )
```

Comment:
I hope no test uses `thunder.compile` and fail on these lines. Let's see.

Reply:
```
FAILED thunder/tests/test_nvfuser.py::test_rematerialization_with_forward_and_backward_from_trace_torch_cpu_None - NotImplementedError: torch.autograd.Function integration is not supported in thunder.compile(). Please use thunder.jit() to compile functions that require torch.autograd.Function.
FAILED thunder/tests/test_grad.py::test_jvp_correctness_unsqueeze_torch_cpu_float64 - NotImplementedError: torch.autograd.Function integration is not supported in thunder.compile(). Please use thunder.jit() to compile functions that require torch.autograd.Function.
FAILED thunder/tests/test_grad.py::test_forward_and_backward_from_trace_torch_cpu_None - NotImplementedError: torch.autograd.Function integration is not supported in thunder.compile(). Please use thunder.jit() to compile functions that require torch.autograd.Function.
FAILED thunder/tests/test_grad.py::test_jvp_correctness_add_torch_cpu_float64 - NotImplementedError: torch.autograd.Function integration is not supported in thunder.compile(). Please use thunder.jit() to compile functions that require torch.autograd.Function.
FAILED thunder/tests/test_grad.py::test_jvp_correctness_mul_torch_cpu_float64 - NotImplementedError: torch.autograd.Function integration is not supported in thunder.compile(). Please use thunder.jit() to compile functions that require torch.autograd.Function.
FAILED thunder/tests/test_grad.py::test_jvp_correctness_sin_torch_cpu_float64 - NotImplementedError: torch.autograd.Function integration is not supported in thunder.compile(). Please use thunder.jit() to compile functions that require torch.autograd.Function.
FAILED thunder/tests/test_grad.py::test_jvp_correctness_broadcast_in_dim_torch_cpu_float64 - NotImplementedError: torch.autograd.Function integration is not supported in thunder.compile(). Please use thunder.jit() to compile functions that require torch.autograd.Function.
```

Reply:
Yeah, we still have some of these, but switching them to thunder.jit would be very much appreciated.